### ISSUE TITLE
Sign in first with e-mail/password, then with Microsoft fails silently

### ISSUE THREAD

**Issue by rbubley** [firebase/firebaseui-web#956]
### [REQUIRED] Describe your environment

  * Operating System version: MacOS 12.3
  * Browser version: Chrome Version 100.0.4896.60 (Official Build) (arm64)
  * Firebase UI version: Using the Firebase demo app: https://fir-ui-demo-84a6c.firebaseapp.com/
  * Firebase SDK version: 9.1.3 (Using the Firebase demo app: https://fir-ui-demo-84a6c.firebaseapp.com/)

#### Steps to reproduce:

1. Log in as a new user using email and password.
2. Log out
3. Log in using same e-mail with a Microsoft account.

Page returned says "FirebaseUI Demo
You are signed out." - No error messages, no hint what the problem is.

Expected behaviour:
 Window saying "You already have an account", with a button to take you to the correct provider.

(And a further, presumably related issue:
4. Go back to sign in, and select sign in with google. 

Result: firebaseui rectangle appears, with the blue animated loading bar at the top, and just stays there - endlessly waiting for something.
)

This issue might be related to #954

**Comment by Jerit3787** [comment]
Is this have any response from the team? I recently switched to v9 and this broke as well. For me, it was reporting (auth/account-exists-with-different-credential). Wasn't firebase-ui should handle properly linking accounts? In v8 still works but broken in v9. Used latest version of firebase and firebase-ui. In the demo it show nothing at all.

**Comment by jekh** [comment]
I just came looking for a solution to this as well. I've also tried logging in with a Microsoft account with the same email as an account registered using the Google provider (rather than email) -- the same thing occurs. Neither of the success or failure callbacks are triggered -- `signInSuccessWithAuthResult` and `signInFailure`. It just fails silently.

It does seem like firebaseui should either prompt linking the Microsoft account to the existing credential, or suggest logging in with the other provider instead (similar to when you attempt to sign in with email for an OAuth-linked account).

**Comment by Jerit3787** [comment]
Yeah, I do wish it had a fallback. But so far I’ve tested if an account is registered first with an microsoft account, then tries to registers with an email and password. The menu works out of the box. I guess there’s issue behind this when using microsoft as a secondary auth provider for an account.

**Comment by cgil** [comment]
Is there any solution to merging accounts across providers?

**Comment by dcoleyoung** [comment]
The correct behavior exists for the email/password, e.g. Account exists use google for example.  Complete hang on FB login.

**Comment by camnewnham** [comment]
This still occurs with 
```
    "firebase": "^9.18.0",
    "firebaseui": "^6.0.2",
```

Login with Google OK
Login with email/pass OK (redirects to login with Google)
Login with Microsoft silently fails

It creates a `firebaseui:pendingEmailCredential` in session storage but then nothing.

**Comment by killianhuyghe** [comment]
I "fixed" this by reverting to firebase v8 and firebaseui v5 as suggested in this [issue](https://github.com/firebase/firebaseui-web/issues/960).

As I am using firebase v9 for almost everything else in my app, my workaround implies dynamically loading firebase v8 only when I need to show the sign in view (so that I don't pay the cost of double loading firebase every time).

Note that the thread also contains another workaround that should be compatible with firebase v9 (didn't test), but it seems quite hackish.

**Comment by michjun** [comment]
seems like this is still broken as of today. Please at least fire signInFailure callback so that we can do someworkarounds

**Comment by dackers86** [comment]
Thank you for taking the time to submit this issue.
A new release is planned shortly, which should resolve this request. 

In the meantime, we will close this  bug, if this is not resolved - please feel free to create a new issue/feature request/PR against the new version.
